tests/EdFi.Security.DataAccess.UnitTests/Contexts/SecurityContextFactoryTests.cs

```diff
@@ -18,16 +19,17 @@ public class SecurityContextFactoryTests
         [Test]
         public void Given_null_argument_then_constructor_throws_exception()
         {
-            Should.Throw<ArgumentNullException>(() => new SecurityContextFactory(null));
+            Should.Throw<ArgumentNullException>(() => new SecurityContextFactory(null,null));
         }
 
         [Test]
         public void Given_configured_for_SqlServer_then_create_SqlServerSecurityContext()
         {
-            var configurationProvider = A.Fake<IApiConfigurationProvider>();
-            A.CallTo(() => configurationProvider.DatabaseEngine).Returns(DatabaseEngine.SqlServer);
-
-            new SecurityContextFactory(configurationProvider)
+            var connectionstringProvider = A.Fake<ISecurityDatabaseConnectionStringProvider>();
```

Comment:
The tests are failing, paritally as you cannot fact the database engine as it is an enum. You also need to return the connection string for the provider, (look at the web api connection string in web.base.config and web.npgsql.config.

Here is a sample of what the test should look like.
```
        public void Given_configured_for_Postgres_then_create_PostgresSecurityContext()
        {
            var connectionstringProvider = A.Fake<ISecurityDatabaseConnectionStringProvider>();

            A.CallTo(() => connectionstringProvider.GetConnectionString()).Returns(
                "Host=localhost; Port=5432; Username=postgres; Database=EdFi_Security; Application Name=EdFi.Ods.WebApi;");

            new SecurityContextFactory(connectionstringProvider, DatabaseEngine.Postgres)
                .CreateContext()
                .ShouldBeOfType<PostgresSecurityContext>()
                .ShouldNotBeNull();
        }
```